Desktop/widgets/textareabase.cpp

```diff
@@ -78,7 +78,8 @@ void TextAreaBase::rScriptDoneHandler(const QString & result)
 		setHasScriptError(false);
 		setProperty("infoText", tr("Model applied"));
 
-		_boundControl->setBoundValue(Json::Value(text().toStdString()));
+		_boundControl->rScriptDoneHandler(result);
+
```

Comment:
result is here always empty, so you don't have to send it.
I think it would be more logical to make the check whether the result is OK in the rScriptDoneHandler method (the boundcontrol class makes up the R script, so it knows whether the result is OK), and that this method returns a boolean or better the error message, so that the TextArea class can handle the error:
```
void TextAreaBase::rScriptDoneHandler(const QString & result)
{
	QString error = _boundControl->rScriptDoneHandler(result)
	if (error.isEmpty())
	{
		setHasScriptError(false);
		setProperty("infoText", tr("Model applied"));
	}
	else
	{
		setHasScriptError(true);
		setProperty("infoText", error);
	}
}

```

And in rScriptDoneHandler:
```
QString BoundControlLavaanTextArea::rScriptDoneHandler(const QString &result)
{
  if (!result.isEmpty()) return result;
....
```